website/blog/2018-06-25-on-consuming-and-publishing-es6+-packages.md

```diff
@@ -0,0 +1,218 @@
+
+#### Staging Process
+
+- The [TC39 staging process](https://tc39.github.io/process-document/) does not always move forward: there is the possibility for a proposal to move to any point in the process: even moving backwards from Stage 3 to Stage 2 as the case with Numeric Separators (`1_000`) or dropped entirely (`Object.observe()`, and others you may have forgotten)
+- It can be hard to differentitate between the stages if you aren't involved: most people would advise that Stage 0-2 is unstable though.
+- Summary: Stage 0 has no criteria and is just an idea, Stage 1 is accepting that the problem is worth solving, Stage 2 is about describing a solution in spec text, Stage 3 means the specific solution is thought out, and Stage 4 means that it is ready for inclusion with tests, multiple browser implementations, and in-the-field experience.
+
+#### Publishing Non-standard Syntax
+
+So actually publishing it in a library may cause even more issues. Because a TC39 proposal (even at Stage 3) has a possibility of changing, it means you will inevitability have to change the code.
+
+At least if you ship the compiled version, it will still work and the library maintainer can change the output so that it works the same as before. Shipping the uncompiled version means that anyone consuming a package will necessarily have to have a build step to use it, and will have to have the same version of Babel as you. This is no different than using TS/JSX/Flow which you wouldn't publish either.
+
+TODO: Or a proposal might just stall because people are busy, etc. Decorators are a great example of this.
+
+### Conflating ESM and ES6+
+
+When you do `import foo from "foo"` or `require("foo")` and `foo` doesn't have an `index.js`, it will resolve to the `main` field in the `package.json` of the module.
+
+Some tools like Rollup/Webpack will also read from another field called `module` (previously `jsnext:main`). It uses this to instead resolve to the ESM file.
+
+- An example with [`redux`](https://github.com/reactjs/redux)
+
+```js
+// redux package.json
+{
+  ...
+  "main": "lib/redux.js", // ES5 + CJS
+  "module": "es/redux.js", // ES5 + ESM
+}
+```
+
+This was introduced so that users could consume ES2015 modules (ESM).
+
+However, the sole intention of this field is ESM, not anything else. The [Rollup docs](https://github.com/rollup/rollup/wiki/pkg.module#wait-it-just-means-import-and-export--not-other-future-javascript-features) make it clear that it's not intended for future JavaScript syntax.
+
+What this means is that we may need another way to handle ES6+ even though some libraries may choose to do this anyway.
+
+A common suggestion is for libraries to start publishing ES2015 under another field like `es2015`, e.g. `"es2015": "es2015/redux.js"`.
```

Comment:
What follows are some non-scalable solutions to the problem described above. As such, you could add a `####`-level heading here to separate the problem from the (bad) solutions.